### ISSUE TITLE
Crash when compiling quasar project with cypress

### ISSUE THREAD

**Issue by dinuta** [oven-sh/bun#11042]
### How can we reproduce the crash?

{
  "name": "MyUIApp",
  "version": "1.0.9",
  "description": "Quasar service with cypress.",
  "productName": "UI",
  "author": "Dinuta",
  "private": false,
  "scripts": {
    "lint": "eslint --ext .js,.vue ./",
    "test": "cypress run",
    "cypress:run": "cypress run",
    "cypress:open": "cypress open"
  },
  "dependencies": {
    "@quasar/extras": "^1.16.11",
    "@vitejs/plugin-vue": "^5.0.4",
    "axios": "^1.6.8",
    "cordova-android": "^12.0.1",
    "cordova-common": "^5.0.0",
    "core-js": "^3.37.0",
    "dotenv": "^16.4.5",
    "lodash": "^4.17.21",
    "qs": "^6.12.1",
    "quasar": "^2.16.1",
    "uuid": "^3.4.0",
    "vue": "^3.4.27",
    "vue-i18n": "^9.13.1",
    "vue-router": "^4.3.2",
    "vuedraggable": "^4.1.0",
    "vuex": "^4.1.0"
  },
  "devDependencies": {
    "@babel/core": "^7.24.5",
    "@babel/eslint-parser": "^7.24.1",
    "@cypress/vue": "^4.2.2",
    "@quasar/app-vite": "^1.9.2",
    "@quasar/babel-preset-app": "^2.0.3",
    "@testing-library/cypress": "^10.0.1",
    "autoprefixer": "^10.4.19",
    "cypress": "^13.9.0",
    "eslint": "^8.57.0",
    "eslint-config-prettier": "^9.1.0",
    "eslint-plugin-vue": "^9.25.0",
    "postcss": "^8.4.38",
    "prettier": "^3.2.5",
    "typescript": "^5.4.5"
  },
  "browserslist": [
    "last 10 Chrome versions",
    "last 10 Firefox versions",
    "last 4 Edge versions",
    "last 7 Safari versions",
    "last 8 Android versions",
    "last 8 ChromeAndroid versions",
    "last 8 FirefoxAndroid versions",
    "last 10 iOS versions",
    "last 5 Opera versions"
  ],
  "engines": {
    "node": ">= 12.22.1",
    "npm": ">= 6.13.4",
    "yarn": ">= 1.21.1",
    "bun": ">= 1.1.8"
  }
}

### Relevant log output

```shell
PS C:\Users\dinuta\IdeaProjects\quasar> bun install
[0.00ms] ".env"
bun install v1.1.8 (89d25807)
warn: incorrect peer dependency "vite@2.9.18"
  📦 Installing [45/849] ============================================================
Bun v1.1.8 (89d25807) Windows x64
Args: "C:\Users\dinuta\.bun\bin\bun.exe", "install"
Features: dotenv extracted_packages(3) 
Elapsed: 1911ms | User: 375ms | Sys: 5875ms
RSS: 0.18GB | Peak: 0.19GB | Commit: 0.29GB | Faults: 50431

panic(main thread): index out of bounds
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

To send a redacted crash report to Bun's team,
please file a GitHub issue using the link below:

 https://bun.report/1.1.8/wi189d2580AgoBk+luH6sruHuo7nE63zkE0jnjE8zqvFqg4BiiJojr+9CCa\KERNEL32.DLLo05FA0eNrLzEtJrVDILy1RyE9TSMovzUspBgBHeAcx
```

### Stack Trace (bun.report)

Bun v1.1.8 ([`89d2580`](https://github.com/oven-sh/bun/tree/89d25807fbf4488d3bfcb116484d0c2cf824f98c)) on windows x86_64 [InstallCommand]

**panic**: index out of bounds

- [`install.zig:4127`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/install/install.zig#L4127): `enqueueDependencyWithMain`
- [`install.zig:4967`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/install/install.zig#L4967): `processDependencyList`
- [`install.zig:5687`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/install/install.zig#L5687): `runTasks`
- [`install.zig:10170`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/install/install.zig#L10170): `installPackages`
- [`filter_run.zig:428`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/cli/filter_run.zig#L428): `install`
- [`cli.zig:1440`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/cli.zig#L1440): `start`
- [`cli.zig:62`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/cli.zig#L62): `start`
- [`main.zig:51`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/main.zig#L51): `main`
- [`exe_common.inl:78`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/src/startup/exe_common.inl#L78): `invoke_main`
- ??? at `0x17344` in KERNEL32.DLL

**Comment by Jarred-Sumner** [comment]
Duplicate of #9808

**Comment by Jarred-Sumner** [comment]
Closing as duplicate, we will fix this but please follow along in #9808
